src/WebJobs.Script/Host/ProxyFunctionMetadata.cs

```diff
@@ -0,0 +1,19 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using Microsoft.Azure.WebJobs.Script.Abstractions.Description;
+using Microsoft.Azure.WebJobs.Script.Description;
+
+namespace Microsoft.Azure.WebJobs.Script
+{
+    public class ProxyFunctionMetadata : FunctionMetadata
```

Comment:
With this new Type, can we now remove the FunctionMetadata.IsProxy property from the public surface area?

Reply:
I think it may still help when we filter proxy functions in some places. I can replace it with `if (metadata is ProxyFunctionMetadata)` which adds a slight more overhead (I think?).